### PR TITLE
Update api-server XCCDF and OVAL for ocp4-isms

### PR DESCRIPTION
**Pull request by redhatrises**
- Update some of the api-server XCCDF and OVAL for ocp4-isms

### CODE REVIEW

File: applications/openshift/api-server/api_server_audit_log_maxage/rule.yml [hunk 1]
```diff
@@ -6,6 +6,16 @@ title: 'Configure the Audit Log Maximum Retention Days (maximumFileRetentionDays
 
 description: |-
     To configure audit log retention, edit the API Server pod specification file
+{{%- if product == "ocp4" %}}
+    <tt>/etc/kubernetes/static-pod-resources/configmaps/config/config.yaml</tt> on the master node(s) and set
+    the <tt>maximumFileRetentionDays</tt> parameter to 30 or as appropriate number of days:
```
Comment (shawndwells) [inline applications/openshift/api-server/api_server_audit_log_maxage/rule.yml]:
```suggestion
    the <tt>maximumFileRetentionDays</tt> parameter to <tt>30</tt>, or the appropriate number of days:
```

File: applications/openshift/api-server/api_server_audit_log_maxsize/rule.yml [hunk 2]
```diff
@@ -6,6 +6,17 @@ title: 'Configure Maximum Audit Log Size'
 
 description: |-
     To rotate audit logs upon reaching a maximum size, edit the API Server pod
+{{%- if product == "ocp4" %}}
+    specification file <tt>/etc/kubernetes/static-pod-resources/configmaps/config/config.yaml</tt> on
+    the master node(s) and set the <tt>maximumFileSizeMegabytes</tt> parameter to
+    an appropriate size in MB. For example, to set it to 100 MB:
+    <pre>
+    "auditConfig":{
```
Comment (shawndwells) [inline applications/openshift/api-server/api_server_audit_log_maxsize/rule.yml]:
```suggestion
    "auditConfig": {
```

File: applications/openshift/api-server/api_server_audit_log_maxsize/rule.yml [hunk 3]
```diff
@@ -6,6 +6,17 @@ title: 'Configure Maximum Audit Log Size'
...
+      ...
+      "maximumFileSizeMegabytes":100,
+      ...
```
Comment (shawndwells) [inline applications/openshift/api-server/api_server_audit_log_maxsize/rule.yml]:
```suggestion
      ...
      },
```

File: applications/openshift/api-server/api_server_audit_log_maxbackup/rule.yml [hunk 4]
```diff
@@ -6,6 +6,17 @@ title: 'Configure the Maximum Retained Audit Logs'
...
     To configure how many rotations of audit logs are retained, edit the API Server
+{{%- if product == "ocp4" %}}
+    pod specification file <tt>/etc/kubernetes/static-pod-resources/configmaps/config/config.yaml</tt>
+    on the master node(s) and set the <tt>maximumRetainedFiles</tt> parameter to
+    <tt>10</tt> or to an organizationally appropriate value:
+       <pre>
+    "auditConfig":{
+      ...
+      "maximumRetainedFiles":10,
+      ...
```
Comment (shawndwells) [inline applications/openshift/api-server/api_server_audit_log_maxbackup/rule.yml]:
```suggestion
      ...
      },
```

File: applications/openshift/api-server/api_server_anonymous_auth/rule.yml [hunk 5]
```diff
@@ -8,11 +8,24 @@ description: |-
...
+{{%- if product == "ocp4" %}}
+    <tt>/etc/kubernetes/static-pod-resources/configmaps/config/config.yaml</tt> on the master node(s)
+    and set the below parameter:
+    <pre>
+    "apiServerArguments":{
+      ...
+      "anonymous-auth":[
+        "false"
+      ],
+      ...
```
Comment (shawndwells) [inline applications/openshift/api-server/api_server_anonymous_auth/rule.yml]:
```suggestion
      ...
      },
```

File: applications/openshift/api-server/api_server_audit_log_maxage/rule.yml [hunk 6]
```diff
@@ -6,6 +6,16 @@ title: 'Configure the Audit Log Maximum Retention Days (maximumFileRetentionDays
...
+    <pre>
+    "auditConfig":{
+      ...
+      "maximumFileRetentionDays":30,
+      ...
```
Comment (shawndwells) [inline applications/openshift/api-server/api_server_audit_log_maxage/rule.yml]:
```suggestion
      ...
      },
```

File: applications/openshift/api-server/api_server_audit_log_path/rule.yml [hunk 7]
```diff
@@ -7,6 +7,17 @@ title: 'Configure the Audit Log Path'
...
     must be set. Edit the API Server pod specification file
+{{%- if product == "ocp4" %}}
+    <tt>/etc/kubernetes/static-pod-resources/configmaps/config/config.yaml</tt> on the master node(s)
+    and set the <tt>audit-log-path</tt> to a suitable path and file
+    where you would like audit logs to be written. For example:
+    <pre>
+    "auditConfig":{
+      ...
+      "auditFilePath":"/var/log/kube-apiserver/audit.log",
+      ...
```
Comment (shawndwells) [inline applications/openshift/api-server/api_server_audit_log_path/rule.yml]:
```suggestion
      ...
      },
```

File: applications/openshift/api-server/api_server_authorization_mode/rule.yml [hunk 8]
```diff
@@ -9,11 +9,23 @@ description: |-
...
     validate that <tt>authorization-mode</tt> is configured to <tt>Webhook</tt>
+{{%- if product == "ocp4" %}}
+    in <tt>/etc/kubernetes/static-pod-resources/configmaps/config/config.yaml</tt>:
+    <pre>
+    "apiServerArguments":{
+      ...
+      "authorization-mode":[
+        "Webhook"
+      ],
+      ...
```
Comment (shawndwells) [inline applications/openshift/api-server/api_server_authorization_mode/rule.yml]:
```suggestion
      ...
      },
```

File: applications/openshift/api-server/api_server_basic_auth/rule.yml [hunk 9]
```diff
@@ -7,12 +7,25 @@ title: 'Disable basic-auth-file for the API Server'
...
+{{%- if product == "ocp4" %}}
+    server pod specification file <tt>/etc/kubernetes/static-pod-resources/configmaps/config/config.yaml</tt>
+    on the master node and remove the <tt>basic-auth-file</tt> parameter.
+    <pre>
+    "apiServerArguments":{
+      ...
+      "basic-auth-file":[
+        "/path/to/any/file"
+      ],
+      ...
```
Comment (shawndwells) [inline applications/openshift/api-server/api_server_basic_auth/rule.yml]:
```suggestion
      ...
      },
```

File: applications/openshift/api-server/api_server_client_ca/rule.yml [hunk 10]
```diff
@@ -10,12 +10,23 @@ description: |-
...
     that <tt>servingInfo</tt> has the <tt>clientCA</tt> configured in 
     the API Server pod specification file
+{{%- if product == "ocp4" %}}
+    <tt>/etc/kubernetes/static-pod-resources/configmaps/config/config.yaml</tt> on the master
+    node(s) to something similar to:
+    <pre>
+    "servingInfo":{
+      ...
+      "clientCA":"/etc/kubernetes/static-pod-certs/configmaps/client-ca/ca-bundle.crt",
+      ...
```
Comment (shawndwells) [inline applications/openshift/api-server/api_server_client_ca/rule.yml]:
```suggestion
      ...
      },
```

File: applications/openshift/api-server/api_server_etcd_ca/rule.yml [hunk 11]
```diff
@@ -8,6 +8,18 @@ description: |-
...
+{{%- if product == "ocp4" %}}
+    that <tt>storageConfig</tt> has the <tt>ca</tt> configured in 
+    the API Server pod specification file
+    <tt>/etc/kubernetes/static-pod-resources/configmaps/config/config.yaml</tt> on the master
+    node(s) to something similar to:
+    <pre>
+    "storageConfig":{
+      ...
+      "ca":"/etc/kubernetes/static-pod-resources/configmaps/etcd-serving-ca/ca-bundle.crt",
+      ...
```
Comment (shawndwells) [inline applications/openshift/api-server/api_server_etcd_ca/rule.yml]:
```suggestion
      ...
      },
```

File: applications/openshift/api-server/api_server_etcd_ca/rule.yml [hunk 12]
```diff
@@ -29,13 +42,19 @@ severity: medium
...
     cis: 1.2.32
 
-ocil_clause: '<tt>ca</tt> is not set as appropriate for <tt>etcdClientInfo</tt>'
+ocil_clause: '{{%- if product == "ocp4" %}}<tt>ca</tt> is not set as appropriate for <tt>etcdClientInfo</tt>{{% else %}}<tt>ca</tt> is not set as appropriate for <tt>storageConfig</tt>{{%- endif %}}'
 
 ocil: |-
     Run the following command on the master node(s):
+{{%- if product == "ocp4" %}}
+    <pre>$ sudo grep ca /etc/kubernetes/static-pod-resources/configmaps/config/config.yaml</pre>
+    The output should contain something similar to:
```
Comment (shawndwells) [inline applications/openshift/api-server/api_server_etcd_ca/rule.yml]:
```suggestion
    The output should be similar to:
```

File: applications/openshift/api-server/api_server_admission_control_plugin_AlwaysAdmit/oval/shared.xml [hunk 13]
```diff
@@ -17,10 +17,17 @@
     <ind:object object_ref="object_api_server_admission_control_plugin_AlwaysAdmit" />
   </ind:textfilecontent54_test>
 
+{{%- if product == "ocp4" %}}
+  <ind:textfilecontent54_object id="object_api_server_admission_control_plugin_AlwaysAdmit" version="1">
+    <ind:filepath>/etc/kubernetes/static-pod-resources/configmaps/config/config.yaml</ind:filepath>
+    <ind:pattern operation="pattern match">^.*"serviceAccountPublicKeyFiles":[\s]*\[.*"(\S+)".*\][,]*.*$</ind:pattern>
```
Comment (shawndwells) [inline applications/openshift/api-server/api_server_admission_control_plugin_AlwaysAdmit/oval/shared.xml]:
```suggestion
    <ind:pattern operation="pattern match">^.*"serviceAccountPublicKeyFiles":[\s]*\[.*"(\S+)".*\]*.*$</ind:pattern>
```

Comment (shawndwells) [reply applications/openshift/api-server/api_server_admission_control_plugin_AlwaysAdmit/oval/shared.xml]: Should the regex evaluate for the final ``,``?

eg if this ends up as the last option in the stanza, the ``,`` will not be present.

File: applications/openshift/api-server/api_server_audit_log_maxsize/rule.yml (hunk 2 above)
Comment (redhatrises) [reply applications/openshift/api-server/api_server_audit_log_maxsize/rule.yml]: Going to keep the original syntax considering indentation was not done in the config.

File: applications/openshift/api-server/api_server_admission_control_plugin_AlwaysAdmit/oval/shared.xml (hunk 13 above)
Comment (redhatrises) [reply applications/openshift/api-server/api_server_admission_control_plugin_AlwaysAdmit/oval/shared.xml]: Not sure it makes sense for us to evaluate that atm. We can revisit later if it does.

File: applications/openshift/api-server/api_server_etcd_ca/rule.yml (hunk 12 above)
Comment (redhatrises) [reply applications/openshift/api-server/api_server_etcd_ca/rule.yml]: I would like to keep the original language considering how it might and can change.

**Pull request merged**